velociraptor/autoplotter/objects.py

```diff
@@ -872,11 +877,8 @@ def make_plot(
 
             self._add_shading_to_axes(ax)
 
-            for data, bracket in zip(
-                self.observational_data, self.observational_data_redshift_bracketing
-            ):
-                if bracket[0] < catalogue.z and bracket[1] > catalogue.z:
-                    data.plot_on_axes(ax, errorbar_kwargs=dict(zorder=-10))
+            for data in valid_observational_data:
+                data.plot_on_axes(ax, errorbar_kwargs=dict(zorder=-10))
```

Comment:
What if observational-data instances are highly concentrated in the redshift domain (e.g. if we have a fitting function that gives us the data at any z we want)? In this case,  `valid_observational_data` array might become very large and we will have too many observational-data curves in the plot.

Don't we want to impose **a limit** on the number of observational-data curves to be plotted? (and choose those whose z's are the closest to the snapshot's z) ?

Reply:
That's now done on a dataset-by-dataset basis.